daml-foundations/daml-ghc/daml-prim-src/GHC/Enum.daml

```diff
@@ -69,61 +66,59 @@ class  Bounded a  where
-    --   a possible implementation being `enumFrom n = n : enumFrom (succ n)`.
-    --   For example:
+    -- | Return a list of the `Enum` values starting at the `Int`
+    -- position. For example:
     --
-    --     * `enumFrom 4 :: [Integer] = [4,5,6,7,...]`
-    --     * `enumFrom 6 :: [Int] = [6,7,8,9,...,maxBound :: Int]`
+    --     * `enumFrom 6 : [Int] = [6,7,8,9,...,maxBound : Int]`
     enumFrom            : a -> [a]
-    -- | Used in Haskell's translation of `[n,n'..]`
-    --   with `[n,n'..] = enumFromThen n n'`, a possible implementation being
-    --   `enumFromThen n n' = n : n' : worker (f x) (f x n')`,
-    --   `worker s v = v : worker s (s v)`, `x = fromEnum n' - fromEnum n` and
-    --   `f n y
-    --     | n > 0 = f (n - 1) (succ y)
-    --     | n < 0 = f (n + 1) (pred y)
-    --     | otherwise = y`
-    --   For example:
+    -- | Returns a list of the `Enum` values with the first value at
+    -- the first `Int` position, the second value at the second `Int`
+    -- position, and further values with the same distance between them.
+    --
+    -- For example:
     --
-    --     * `enumFromThen 4 6 :: [Integer] = [4,6,8,10...]`
+    --     * `enumFromThen 4 6 :: [Int] = [4,6,8,10...]`
     --     * `enumFromThen 6 2 :: [Int] = [6,2,-2,-6,...,minBound :: Int]`
     enumFromThen        : a -> a -> [a]
-    -- | Used in Haskell's translation of `[n..m]` with
-    --   `[n..m] = enumFromTo n m`, a possible implementation being
-    --   `enumFromTo n m
-    --      | n <= m = n : enumFromTo (succ n) m
-    --      | otherwise = []`.
-    --   For example:
+    -- | Returns a list of the `Enum` values with the first value at
+    -- the first `Int` position, and the last value at the last `Int`
+    -- position.
+    --
+    -- For example:
     --
     --     * `enumFromTo 6 10 :: [Int] = [6,7,8,9,10]`
-    --     * `enumFromTo 42 1 :: [Integer] = []`
     enumFromTo          : a -> a -> [a]
-    -- | Used in Haskell's translation of `[n,n'..m]` with
-    --   `[n,n'..m] = enumFromThenTo n n' m`, a possible implementation
-    --   being `enumFromThenTo n n' m = worker (f x) (c x) n m`,
-    --   `x = fromEnum n' - fromEnum n`, `c x = bool (>=) (<=) (x > 0)`
-    --   `f n y
-    --      | n > 0 = f (n - 1) (succ y)
-    --      | n < 0 = f (n + 1) (pred y)
-    --      | otherwise = y` and
-    --   `worker s c v m
-    --      | c v m = v : worker s c (s v) m
-    --      | otherwise = []`
-    --   For example:
+    -- | Returns a list of the `Enum` values with the first value at
+    -- the first `Int` position, the second value at the second `Int`
+    -- position, and further values with the same distance between them,
+    -- with the final value at the final `Int` position.
+    --
+    -- For example:
     --
-    --     * `enumFromThenTo 4 2 -6 :: [Integer] = [4,2,0,-2,-4,-6]`
+    --     * `enumFromThenTo 4 2 -6 :: [Int] = [4,2,0,-2,-4,-6]`
```

Comment:
Use `:` for type signatures here

Reply:
It would be so good if we could actually test these for correctness